oozie-to-airflow/tests/test_templates.py

```diff
@@ -41,26 +64,61 @@ def mutate(parent: Dict[str, Any], mutations: Dict[str, Any]):
     return result
 
 
-def get_by_path(target, segments):
+def get_value_by_path(target: Any, path: List[Union[str, int]]) -> Any:
+    """
+    Gets the value at path of dict or list.
+
+    :Example:
+
+    .. code-block:: pycon
+
+        In [0]: target = { 'a': { 'b': { 'c': 3 } } }
+        Out[0]: {'a': [{'b': {'c': 'AAA'}}]}
+
+        In [1]: target = { 'a': [{ 'b': { 'c': 3 } }] }
+
+        In [2]: get_value_by_path(target, ["a", 0, "b", "c"])
+        Out[2]: 3
+
+    The behavior of the function is similar to:
+    https://lodash.com/docs#get
+    """
     result = target
-    for segment in segments:
+    for segment in path:
         if isinstance(result, dict):
             result = result[segment]
         elif isinstance(result, list):
             result = result[int(segment)]
         else:
-            raise Exception(f"Invalid path: {'.'.join(segments)}")
+            raise Exception(f"Invalid path: {path}")
     return result
 
 
-def set_by_path(target, segments, value):
-    result = get_by_path(target, segments[:-1])
+def set_value_by_path(target: Any, segments: List[Union[str, int]], value: Any) -> None:
```

Comment:
It's way more readable now. Way, way better. Just a nit here - please also change the name to 'path' here

Reply:
I updated a PR. Thanks.